pageserver/src/tenant.rs

```diff
@@ -766,6 +766,10 @@ impl Tenant {
     async fn load(self: &Arc<Tenant>) -> anyhow::Result<()> {
         info!("loading tenant task");
 
+        fail::fail_point!("before-loading-tenant", |_| {
```

Comment:
In `test_pageserver_restart.py`, you're using this with the `sleep(5000)` action.
The `sleep` action is a synchronous `std::thread::sleep`.
You should use the new `failpoint_sleep_millis_async` instead.